browser/android/youtube_script_injector/brave_youtube_script_injector_native_helper.cc

```diff
@@ -0,0 +1,131 @@
+        }
+      });
+
+      var fullscreenBtn = document.querySelector(fullscreenButtonSelector);
+      var videoPlayer = document.querySelector(videoPlaySelector);
+      // Check if fullscreen button and video are available.
+      if (fullscreenBtn && videoPlayer) {
+       delayedPlayAndClick(fullscreenBtn, videoPlayer);
+      } else {
+        // When fullscreen button is not available
+        // clicking the movie player resume the UI.
+        var moviePlayer = document.getElementById("movie_player");
+        var playerContainer = document.getElementById("player-container-id");
+        if (moviePlayer && playerContainer) {
+          // Auto-disconnect the observer after 30 seconds,
+          // a reasonable duration picked after some testing.
+          observerTimeout = setTimeout(() => {
+            observer.disconnect();
+          }, 30000);
+          // Start observing the DOM.
+          observer.observe(playerContainer, { childList: true, subtree: true });
+          // Make sure the player is in focus or responsive.
+          moviePlayer.click();
+        }
+      }
+    }
+  }
+
+  // Click the fullscreen button and play the video and after a delay
+  // to ensure the video is ready.
+  // This is necessary because sometimes (rarely) when switching to fullscreen
+  // mode a video might be paused automatically from the backend if the buffer
+  // was not ready.
+  // The delay allows the video to load properly before attempting to play it.
+  // This is especially important for high quality videos, which may require
+  // some time to buffer before they can be played.
+  // The delay is set to 500 milliseconds, which is a reasonable delay for
+  // the videos to be ready for playback.
+  function delayedPlayAndClick(fullscreenBtn, videoPlayer) {
+    setTimeout(() => {
+      videoPlayer.play();
+    }, 500);
+    fullscreenBtn.click();
+  }
+
+  if (document.readyState === "loading") {
+    // Loading hasn't finished yet.
+    document.addEventListener("DOMContentLoaded", triggerFullscreen);
+  } else {
+    // `DOMContentLoaded` has already fired.
+    triggerFullscreen();
+  }
+}());
+)";
+
+}  // namespace
+
+namespace youtube_script_injector {
+
+// static
+jboolean JNI_BraveYouTubeScriptInjectorNativeHelper_IsYouTubeVideo(
+    JNIEnv* env,
+    const base::android::JavaParamRef<jobject>& jweb_contents) {
+  content::WebContents* web_contents =
+      content::WebContents::FromJavaWebContents(jweb_contents);
+
+  auto url = web_contents->GetLastCommittedURL();
+  return YouTubeScriptInjectorTabHelper::IsYouTubeVideo(url);
+}
+
+// static
+void JNI_BraveYouTubeScriptInjectorNativeHelper_SetFullscreen(
+    JNIEnv* env,
+    const base::android::JavaParamRef<jobject>& jweb_contents) {
+  content::WebContents* web_contents =
+      content::WebContents::FromJavaWebContents(jweb_contents);
+  SetFullscreen(web_contents);
+}
+
+void SetFullscreen(content::WebContents* web_contents) {
```

Comment:
There's no need for JNI here because it's already exposed through mojo so you can call it directly in java